src/main/java/com/grafana/opentelemetry/Log4jConfig.java

```diff
@@ -0,0 +1,44 @@
+package com.grafana.opentelemetry;
+
+import io.opentelemetry.instrumentation.log4j.appender.v2_17.OpenTelemetryAppender;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.core.Appender;
+import org.apache.logging.log4j.core.LoggerContext;
+import org.apache.logging.log4j.core.config.Configuration;
+import org.apache.logging.log4j.core.config.LoggerConfig;
+import org.apache.logging.log4j.Logger;
+
+
+public class Log4jConfig {
+
+    private static final Logger logger = LogManager.getLogger(Log4jConfig.class);
+
+    static void addLog4jAppender() {
+        LoggerContext context = (LoggerContext) LogManager.getContext(false);
+        Configuration config = context.getConfiguration();
+        boolean found = config.getAppenders().values().stream()
+                .anyMatch(a -> a instanceof io.opentelemetry.instrumentation.log4j.appender.v2_17.OpenTelemetryAppender);
+        if (found) {
+            logger.info("log4j2 OpenTelemetryAppender has already been added");
+            return;
+        }
+
+        logger.info("adding log4j OpenTelemetryAppender");
+        OpenTelemetryAppender appender = OpenTelemetryAppender.builder()
+                .setCaptureExperimentalAttributes(true)
```

Comment:
Just wondering if `captureExperimentalAttributes` should be configurable.  What is the impact of having this as true?

Reply:
the thread name and thread id are added as attributes to the log entry - to be used [here](https://grafana.com/docs/opentelemetry/visualization/loki-data/#common-framework-formats)
It's marked as experimental, because the semantic conventions are not stable.

I think it's rather an implementation details.